magento/pwa-studio
# [DO NOT MERGE] Cypress POC

## Description

POC for writing Functional tests with cypress.

## Related Issue





Closes PWA-1434.

## Acceptance



Team should agree with using Cypress for Venia Functional Tests. Also add any improvements/best practices/Suggestions to below TO-DO list.

## TO-DO - 
1. Create alias for Dom elements in one place and call in scripts as needed.
2. Create utility file for commonly used methods like UID generation.
3. Need to built utility function which can wait until all elements of the page loads before going to next step.
4. Remove Cypress Example tests
4. Create a wiki page with information.

### Verification Stakeholders
PWA Team
### Specification



### Verification Steps


1. Check instructions in README.md under /venia-e2e-tests folder.
2. Verify able to Venia run demo tests /cypress/integration/real-data-tests/ and  /cypress/integration/mock-data-tests/




## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.
